controllers/azurecluster_controller.go

```diff
@@ -248,12 +248,14 @@ func (acr *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 		return reconcile.Result{}, wrappedErr
 	}
 
-	// Set APIEndpoints so the Cluster API Cluster Controller can pull them
-	if azureCluster.Spec.ControlPlaneEndpoint.Host == "" {
-		azureCluster.Spec.ControlPlaneEndpoint.Host = clusterScope.APIServerHost()
-	}
-	if azureCluster.Spec.ControlPlaneEndpoint.Port == 0 {
-		azureCluster.Spec.ControlPlaneEndpoint.Port = clusterScope.APIServerPort()
+	if azureCluster.Spec.ControlPlaneEnabled {
```

Comment:
We need to check the values also when the Control Plane is externally managed, something like this:


```go
if !azureCluster.Spec.ControlPlaneEnabled {
		if azureCluster.Spec.ControlPlaneEndpoint.Host == "" {
			conditions.MarkFalse(azureCluster, infrav1.NetworkInfrastructureReadyCondition, "ExternallyManagedControlPlane", clusterv1.ConditionSeverityInfo, "Waiting for the Control Plane host")

			return reconcile.Result{}, nil
		} else if azureCluster.Spec.ControlPlaneEndpoint.Port == 0 {
			conditions.MarkFalse(azureCluster, infrav1.NetworkInfrastructureReadyCondition, "ExternallyManagedControlPlane", clusterv1.ConditionSeverityInfo, "Waiting for the Control Plane port")

			return reconcile.Result{}, nil
		}
	}
```